references/optical_flow/presets.py

```diff
@@ -0,0 +1,64 @@
+import torch
+import transforms as T
+
+
+class OpticalFlowPresetEval(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+        self.transforms = T.Compose(
+            [
+                T.PILToTensor(),
+                T.ConvertImageDtype(torch.float32),
+                T.Normalize(mean=0.5, std=0.5),  # map [0, 1] into [-1, 1]
+                T.ValidateModelInput(),
+            ]
+        )
+
+    def __call__(self, img1, img2, flow, valid):
+        return self.transforms(img1, img2, flow, valid)
+
+
+class OpticalFlowPresetTrain(torch.nn.Module):
+    def __init__(
+        self,
+        # MaybeRandomResizeAndCrop params
+        crop_size,
+        min_scale=-0.2,
+        max_scale=0.5,
+        stretch_prob=0.8,
+        # AsymmetricColorJitter params
+        brightness=0.4,
+        contrast=0.4,
+        saturation=0.4,
+        hue=0.5 / 3.14,
+        # Random[H,V]Flip params
+        asymmetric_jitter_prob=0.2,
+        do_flip=True,
+    ):
+        super().__init__()
+
+        transforms = [
+            T.PILToTensor(),
+            T.AsymmetricColorJitter(
+                brightness=brightness, contrast=contrast, saturation=saturation, hue=hue, p=asymmetric_jitter_prob
+            ),
+            T.RandomApply([T.RandomErase()], p=0.5),
+            T.MaybeResizeAndCrop(
+                crop_size=crop_size, min_scale=min_scale, max_scale=max_scale, stretch_prob=stretch_prob
+            ),
+        ]
+
+        if do_flip:
+            transforms += [T.RandomHorizontalFlip(p=0.5), T.RandomVerticalFlip(p=0.1)]
+
+        transforms += [
+            T.ConvertImageDtype(torch.float32),
+            T.Normalize(mean=0.5, std=0.5),  # map [0, 1] into [-1, 1]
```

Comment:
All transforms are dtype-agnostic, except for the current `RandomErase()` which calls `round()`.

I'm keeping the conversion at the end for now, but in the future it would be worth trying moving it at the beginning of the transforms pipeline.

Reply:
As mentioned earlier, I think it makes more sense to eventually support a `RandomErase` that doesn't take as input a normalized image. This will make the transform similar to all other transforms and enable users to visualize its effect easier (now one needs to undo the scaling). But for now, as mentioned above, it might be confusing to users that some Random Erasing transforms are applied prior normalization while other after.

Reply:
OK,  thanks for the input. I'll try to move this at the beginning of the transforms, and change `RandomErase` to replace with 0 instead of the mean (which should now the same).

I'll start some runs on Chairs + Things today and we should get the results tomorrow. If epe is still good I'll keep the changes, and otherwise I'll keep things as-is and will mark a TODO + open an issue to keep track of this progress. Does that sound good?

Reply:
Just to confirm that we mean the same thing, you will:
1. Move the `RandomErase` after `Normalize` similar to how `RandomErasing` works in TorchVision (all other transforms remain where they are as normal):
https://github.com/pytorch/vision/blob/d367a01a18a3ae6bee13d8be3b63fd6a581ea46f/torchvision/transforms/transforms.py#L1619-L1626

2. Use `mean=0` as now it should be equivalent. Perhaps also check if your implementation can reuse `F.erase()` or other parts of `RandomErasing` class.
3. Run `Chairs + Things` to confirm that the `epe` is OK else rollback.

If that's what you mean, then sounds like a plan!

Reply:
Yes, thanks for the correction.